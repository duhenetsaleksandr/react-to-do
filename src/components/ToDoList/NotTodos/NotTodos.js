import React from 'react';
import './NotTodos.scss';
import { FormattedMessage } from 'react-intl';

const NotTodos = () => {
    return (
        <div className="not-todos">
            <FormattedMessage id="not_todos" />
        </div>
    );
};

export default NotTodos;
