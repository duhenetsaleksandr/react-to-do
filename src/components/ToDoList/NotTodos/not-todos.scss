@use '../../../styles/colors';

.not-todos {
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 50px;
    color: colors.$secondary-color;
    padding: 50px;
    border: 1px solid colors.$secondary-color;
    border-radius: 5px;
}
