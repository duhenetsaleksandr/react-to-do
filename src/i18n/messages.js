import { LOCALES } from './locales'

export const messages = {
    [LOCALES.ENGLISH]: {
        copyright: '©Copyright. All rights deserved.',
        button_add: 'Add task',
        button_edit: 'Edit task',
        placeholder: 'Enter new todo',
        not_todos: 'Not found todos!',
    },
    [LOCALES.RUSSIAN]: {
        copyright: '©Копирайт. Все права защищены.',
        button_add: 'Добавить',
        button_edit: 'Редактировать',
        placeholder: 'Введите новую задачу',
        not_todos: 'Задачи не найдены!',
    },
};
