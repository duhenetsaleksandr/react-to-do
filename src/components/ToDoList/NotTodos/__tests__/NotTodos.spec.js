import React from 'react';
import NotTodos from '../NotTodos';
import { IntlProvider } from "react-intl";
import { messages } from "i18n/messages";
import { LOCALES } from "i18n/locales";

describe('should render NotTodos component', () => {
    it('should contain one div with .not-todos class', () => {
        const component = mount(<NotTodos/>, {
            wrappingComponent: IntlProvider,
            wrappingComponentProps: {
                messages: messages[LOCALES.ENGLISH],
                locale: LOCALES.ENGLISH,
                defaultLocale: LOCALES.ENGLISH,
            },
        });
        const wrapper = component.find('div.not-todos');
        expect(wrapper).toHaveLength(1);
    });

    it('should contain text for not todos', () => {
        const component = mount(<NotTodos/>, {
            wrappingComponent: IntlProvider,
            wrappingComponentProps: {
                messages: messages[LOCALES.ENGLISH],
                locale: LOCALES.ENGLISH,
                defaultLocale: LOCALES.ENGLISH,
            },
        });
        const messageId = component.find('div.not-todos').getElement().props.children.props.id;
        expect(messageId).toBe('not_todos');
    });

    it('should render NotTodos component snapshot', () => {
        const component = shallow(<NotTodos/>);
        expect(component).toMatchSnapshot();
    });
});
