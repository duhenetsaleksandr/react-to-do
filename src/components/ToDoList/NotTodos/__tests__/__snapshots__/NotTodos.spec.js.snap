// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotTodos component should render NotTodos component snapshot 1`] = `
<div
  className="not-todos"
>
  <MemoizedFormattedMessage
    id="not_todos"
  />
</div>
`;
